modules/aql/arrow-aql/src/main/kotlin/arrow/aql/Max.kt

```diff
@@ -0,0 +1,26 @@
+package arrow.aql
+
+import arrow.core.ForId
+import arrow.core.Id
+import arrow.core.identity
+import arrow.core.value
+import arrow.typeclasses.Foldable
+import kotlin.math.max
+
+interface Max<F> {
+
+  fun foldable(): Foldable<F>
+
+  infix fun <A, Z> Query<F, A, Z>.max(f: A.() -> Long): Query<ForId, Long, Long> =
+    foldable().run {
+      Query(
+        select = ::identity,
+        from = Id(from.foldLeft(Long.MIN_VALUE) { acc, a -> max(f(a), acc) })
+      )
+    }
+
+  fun Query<ForId, Long, Long>.value(): Long =
```

Comment:
There's a chance you may end up returning `Option<A>` where `A` is the type of the `Order`. This is because you won't have a min/max value to default to when the container is empty.

Either way, returning Long.MIN_VALUE as the min from an empty array seems incorrect.

Reply:
You are totally right, I'll come back with your suggested changes.